pythonPackages.magic-wormhole: 0.10.4 -> 0.10.5

dotlambda opened the pull request:
###### Motivation for this change
magic-wormhole is broken on master

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

/cc @asymmetric

Commit:
pythonPackages.spake2: 0.7 -> 0.8

asymmetric commented:
Could you edit the second commit message and explain why you removed `glibcLocales`?

asymmetric commented on pkgs/development/python-modules/magic-wormhole/default.nix:
```diff
@@ -22,30 +21,25 @@
   postPatch = ''
     sed -i -e "s|'ifconfig'|'${nettools}/bin/ifconfig'|" src/wormhole/ipaddrs.py
     sed -i -e "s|if (os.path.dirname(os.path.abspath(wormhole))|if not os.path.abspath(wormhole).startswith('/nix/store') and (os.path.dirname(os.path.abspath(wormhole))|" src/wormhole/test/test_cli.py
-  '' + lib.optionalString (pythonAtLeast "3.3") ''
```
Also explaining why this was removed would be nice.

dotlambda replied:
Because of https://github.com/warner/magic-wormhole/commit/13b1539409b0afc1e63ad292322d64bd18e35eb0, which happened after the 0.10.5 release as I see right now.
I can however revert that change if you want.

asymmetric commented on pkgs/development/python-modules/magic-wormhole/default.nix:
```diff
@@ -3,7 +3,6 @@
 , fetchPypi
 , pythonAtLeast
 , nettools
-, glibcLocales
```
Could you edit the second commit message and explain why you removed this?

dotlambda replied:
Tests work without it, so why should it be included?

asymmetric replied:
Did you try to run the binaries? Asking because the checkbox isn't ticked. :+1:

dotlambda replied:
No I didn't since I don't use magic-wormhole. I just came across the build failure.
However, I only ever saw `glibcLocales` added to make the tests run by setting something like `LC_ALL`.

asymmetric commented on pkgs/development/python-modules/magic-wormhole/default.nix:
```diff
@@ -22,30 +21,25 @@
 buildPythonPackage rec {
   pname = "magic-wormhole";
-  version = "0.10.4";
-  name = "${pname}-${version}";
```
For my interest, is `name` not needed?

dotlambda replied:
Not anymore for `buildPython*`.

FRidh commented:
The version bump is already in staging, so this shouldn't be merged until staging has been merged into master.

asymmetric commented:
Tests fail for me when running `nix-build -A pythonPackages.magic-wormhole`.

Commit:
pythonPackages.magic-wormhole-transit-relay: init at 0.1.1

Commit:
pythonPackages.magic-wormhole: fix build

Add magic-wormhole-transit-relay dependency and clean up expression.
Order of the buildInputs was changed to match the one in setup.py.

dotlambda commented:
Sorry, I was not careful enough.
Second try. I had to package a new dependency.

dotlambda commented on pkgs/development/python-modules/magic-wormhole/default.nix:
```diff
@@ -2,37 +2,38 @@
+  checkInputs = [ mock magic-wormhole-transit-relay ];
   buildInputs = [ nettools glibcLocales ];
-  propagatedBuildInputs = [ autobahn cffi click hkdf pynacl spake2 tqdm ipaddress humanize pyopenssl service-identity txtorcon ];
+  propagatedBuildInputs = [ spake2 pynacl six attrs twisted autobahn automat hkdf tqdm click humanize ipaddress txtorcon ];
```
Sorry that I changed the order so much. I just wanted the dependency to be in the same order as in `setup.py`.

asymmetric replied:
Thanks for the info. In general though, I think it's a much better idea to have these comments in the commit messages, so that people could find out without having to find this PR.

dotlambda replied:
Changed commit message.

asymmetric commented:
It still fails for me, but not for the CI, so not sure.

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.35175&attempt_id=168ee7e5-52a5-43de-a03c-59872cc4ca85)

<details><summary>Partial log (click to expand)</summary><p>

```
wormhole.test.test_cli.PregeneratedCode.test_slow_text
wormhole.test.test_cli.PregeneratedCode.test_text
wormhole.test.test_cli.PregeneratedCode.test_text_subprocess
wormhole.test.test_cli.PregeneratedCode.test_text_tor
-------------------------------------------------------------------------------
Ran 355 tests in 28.345s

PASSED (skips=20, successes=335)
/nix/store/2r702zqgchn7fprfx95vb5144h6c6625-python2.7-magic-wormhole-0.10.5
/nix/store/h52alg3shyv9qsmcywdfb9rb99a3zcs3-python3.6-magic-wormhole-0.10.5
```
</p></details>

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.35175&attempt_id=39936935-a969-4768-9b50-60ee3460076d)

<details><summary>Partial log (click to expand)</summary><p>

```
wormhole.test.test_cli.PregeneratedCode.test_slow_text
wormhole.test.test_cli.PregeneratedCode.test_text
wormhole.test.test_cli.PregeneratedCode.test_text_subprocess
wormhole.test.test_cli.PregeneratedCode.test_text_tor
-------------------------------------------------------------------------------
Ran 355 tests in 72.673s

PASSED (skips=20, successes=335)
/nix/store/h7pxl15587mna9qwddlxdvxbfmhz2q72-python2.7-magic-wormhole-0.10.5
/nix/store/m0fcb1iv6g0fsd61za3fr2qgc0anqn4k-python3.6-magic-wormhole-0.10.5
```
</p></details>

dotlambda commented:
@asymmetric Can you try again, please?

asymmetric commented:
@dotlambda nothing changed, AFAICT, so I don't think tests results would be different. Also, this being nix, we shouldn't have cases of "works on my machine", so I'd say go ahead and consider the build green, I must be doing something wrong.

dotlambda commented:
Sorry for pinging so many people, I just rebased on master

grahamc commented:
@GrahamcOfBorg eval

FRidh commented on pkgs/development/python-modules/magic-wormhole/transit-relay.nix:
```diff
@@ -0,0 +1,25 @@
+{ lib, buildPythonPackage, fetchPypi, twisted, mock }:
```
this is a separate package and should thus have its own folder

dotlambda commented:
@GrahamcOfBorg build python2.pkgs.magic-wormhole python3.pkgs.magic-wormhole

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.35175&attempt_id=dfe4aab0-e0e3-44c4-88e6-109ea4983faf)

<details><summary>Partial log (click to expand)</summary><p>

```
wormhole.test.test_cli.PregeneratedCode.test_slow_text
wormhole.test.test_cli.PregeneratedCode.test_text
wormhole.test.test_cli.PregeneratedCode.test_text_subprocess
wormhole.test.test_cli.PregeneratedCode.test_text_tor
-------------------------------------------------------------------------------
Ran 355 tests in 45.889s

PASSED (skips=20, successes=335)
/nix/store/n9fx2csyi3yz7xfhvrl6k0vz4s5rrn1h-python2.7-magic-wormhole-0.10.5
/nix/store/n0cnsz9wmc90ssk3gf6vxj6hvf7c72dj-python3.6-magic-wormhole-0.10.5
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.35175&attempt_id=797f9823-a493-473a-bef6-f1aa7cea6c64)

<details><summary>Partial log (click to expand)</summary><p>

```
wormhole.test.test_cli.PregeneratedCode.test_slow_text
wormhole.test.test_cli.PregeneratedCode.test_text
wormhole.test.test_cli.PregeneratedCode.test_text_subprocess
wormhole.test.test_cli.PregeneratedCode.test_text_tor
-------------------------------------------------------------------------------
Ran 355 tests in 19.709s

PASSED (skips=20, successes=335)
/nix/store/nwfcj3hd53a0p5sj33k70y9bvvy12prr-python2.7-magic-wormhole-0.10.5
/nix/store/wvbi7yzljcaygb5bhzy4zl2bidrzs4i5-python3.6-magic-wormhole-0.10.5
```
</p></details>
